### PR TITLE
A6 : Eight Queens Problem

### PR DESCRIPTION
**Pull request by RaaneshKumar**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -3,17 +3,132 @@
...
+   /// <summary>Eight Queens problem</summary>
    internal class Program {
       #region Method ------------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      /// <summary>This method prints all the required solutions</summary>
+      static void Main () {
+         WriteLine ("Press U to print unique solutions.\n" +
+                    "Else print any other key to print all solutions.");
+         bool isUnique = false;
```
Comment (priyarathna) [inline Program.cs]: Can be simplified into 'PlaceQueen (Console.ReadKey (true).Key is ConsoleKey.U, 0);'

File: Program.cs [hunk 2]
```diff
@@ -3,17 +3,132 @@
...
+               var sln = sColumns.ToArray ();
+               if (isUnique ? row == 7 && IsUnique (sln) : row == 7) sAllSlns.Add (sln);
+               else PlaceQueen (isUnique, row + 1);
+            }
+         }
+      }
+
+      /// <summary>Creates a copy of the given array</summary>
+      static void CopySolution (int[] original, int[] copy) {
+         for (int i = 0; i < 8; i++) copy[i] = original[i];
```
Comment (priyarathna) [inline Program.cs]: Can be simplified as 'static void CopySolution (int[] original, int[] copy) => Array.Copy (original, copy, 8);'

File: Program.cs [hunk 3]
```diff
@@ -3,17 +3,132 @@
...
+      }
+
+      /// <summary>Stores the safe positions of a solution in an array
+      /// and stores all the solution arrays in a list</summary>
+      /// <param name="row">Fixing the value of row to go through all columns of that row</param>
+      static void PlaceQueen (bool isUnique, int row) {
+         for (int col = 0; col < 8; col++) {
+            if (IsSafe (row, col)) {
+               sColumns[row] = col;
+               var sln = sColumns.ToArray ();
```
Comment (priyarathna) [inline Program.cs]: 'ToArray' conversion is not needed.

**Review by priyarathna — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 1 above)
Comment (RaaneshKumar) [reply Program.cs]: This is a recursive method (line 118). So replacing it this way would ask the user multiple times to enter the same input.

Comment (priyarathna) [reply Program.cs]: I think it'll not be. Let me know if you've further doubts.

**Review by priyarathna — CHANGES_REQUESTED** [review]

**Review by priyarathna — APPROVED** [review]

**Pull request closed without merging**